docs/manage/CLI.md

```diff
@@ -190,6 +193,107 @@ The `logout` command ends an authenticated CLI session
+| `--name`     |  name of the custom module to be created, updated, or uploaded    |`create`, `update`|true |
+| `--platform`      |  the platform to encode the resulting module binary as   |`upload`|true |
+| `--version`      |  the version of your module to set for this upload  |`upload`|true |
+| `--module`      |  the payload of the module to upload, including the <file>meta.json</file> configuration file and the packaged module as a `tar.gz` file    |`upload`|true |
+
+See [Using the upload command](#using-the-upload-command) for more information on using these arguments with `module upload`.
+
+##### Custom module creation workflow example
+
+To upload a new module to the Viam Registry, you need to `create` it with a new name, `update` it with your customized changes, and `upload` it to the Viam Registry.
+The following example demonstrates this workflow for an example module `my-first-module`:
+
+```sh {class="command-line" data-prompt="$"}
+## Create a new module 'my-first-module' locally, which creates a placeholder `meta.json` file on the local filesystem:
+viam module create --name 'my-first-module' --org_id 'abc'
+
+## Edit the newly-created meta.json file with the custom module-specific configuration:
+vi meta.json
+
+## Update the module with the new configuration:
+viam module update --name 'my-first-module' --org_id 'abc'
+
+## Package the module as a tar.gz archive:
+tar -zcf my-first-module.tar.gz my-first-module
+
+## Upload the new custom module to the Viam Registry:
+viam module upload --org_id 'abc' --platform "darwin/amd64" --version "1.0.0" --module meta.json my-first-module.tar.gz
+```
+
+To later make changes to the module, the workflow is similar:
+
+```sh {class="command-line" data-prompt="$"}
+## Edit the same meta.json file from the previous workflow with the new configuration:
+vi meta.json
+
+## Update the module with the new configuration:
+viam module update --name 'my-first-module' --org_id 'abc'
+
+## Package the module as a tar.gz archive:
+tar -zcf my-first-module.tar.gz my-first-module
+
+## Upload the new custom module to the Viam Registry:
+viam module upload --org_id 'abc' --platform "darwin/amd64" --version "1.0.1" --module meta.json my-first-module.tar.gz
+```
+
+##### Using the upload command
+
+When using `module upload`, consider the following:
+
+* The `--platform` flag only accepts one of the following:
+  * `darwin/amd64` - macOS platforms running Apple Silicon
+  * `darwin/x86_64` - macOS platforms with Intel CPUs
+  * `linux/aarch64` - Linux platforms with the `aarch64` architecture
+  * `linux/x86_64` - Linux platforms with Intel architecture
+* The `--version` flag accepts any string value, and is passed to the Viam Registry without any validation.
+* The `--module` flag must include both the <file>meta.json</file> and also a `tar.gz` archive of your module.
+  Currently, `module upload` only supports `tar.gz` archives.
```

Comment:
Added my best guesses for this section: please correct where I am a poor guesser!